proposals/improved-interpolated-strings.md

```diff
@@ -287,12 +272,106 @@ Do we want to have builders for `System.IFormattable` and `System.FormattableStr
+that were interpolated and the components that were part of the base string.
 
-Both the general pattern and the specific changes for interpolated strings directly converted to `string`s follow the same lowering pattern. The `Create` method is
-invoked on the receiver (whether that's the temporary method receiver for an _implicit\_string\_builder\_conversion_ derived from the applicable function member algorithm, or a
-standard conversion derived from the target type). If the call returned `true`, `Append...` is repeatedly invoked on the builder out parameter, with each part of the interpolated string,
-in order, stopping subsequent calls if a `Append...` call returns `false`. Finally, the original method is called, passing the initialized builder in place of the interpolated string expression.
+**Open Question**
+
+Some scenarios, like structured logging, want to be able to provide names for interpolation elements. For example, today a logging call might look like
+`Log("{name} bought {itemCount} items", name, items.Count);`. The names inside the `{}` provide important structure information for loggers that help with ensuring output
+is consistent and uniform. Some cases might be able to reuse the `:format` component of an interpolation hole for this, but many loggers already understand format specifiers
+and have existing behavior for output formatting based on this info. Is there some syntax we can use to enable putting these named specifiers in?
+
+Some cases may be able to get away with `CallerArgumentExpression`, provided that support does land in C# 10. But for cases that invoke a method/property, that may not be
+sufficient.
+
+#### Lowering
+
+Given an _applicable\_interpolated\_string\_builder\_type_ `T` and an _interpolated\_string\_expression_ `i` that had a valid `Create` method `Fc` and `Append...` methods `Fa` resolved,
+lowering for `i` is performed as follows:
+
+1. If the `InterpolatedStringBuilderAttribute` on `T` specifies false for `UseLazyComputation`, the following steps are taken:
+    1. Any arguments to `Fc` that occur lexically before `i` are evaluated and stored into temporary variables in lexical order.
+    2. All _interpolation_ components of `i` are evaluated in lexical order and stored into temporary variables.
+    3. `Fc` is called with the length of the interpolated string literal components, the number of _interpolation_ holes, any previously evaluated arguments, and an a `bool` out argument
+    (if `Fc` was resolved with one as the last parameter). The result is stored into a temporary value `ib`.
+    4. If `Fc` ended with a `bool` out argument, a check on that `bool` value is generated. If true, `Fa` will be called. Otherwise, they will not be called.
+    5. For every `Fax` in `Fa`, `Fax` is called on `ib` with either the current literal component or the saved temporary variable, as appropriate. If `Fax` returns a `bool`, the result is
+    logically anded with all preceeding `Fax` calls.
```

Comment:
It feels like this conflicts with "specifies false for `UseLazyComputation`" stated above